src/grpc/dp_grpc_impl.c

```diff
@@ -655,6 +681,9 @@ static int dp_process_delete_nat(struct dp_grpc_responder *responder)
 	reply->addr.ip_type = RTE_ETHER_TYPE_IPV4;
 	reply->addr.ipv4 = s_data->nat_ip;
 	dp_del_vip_from_dnat(s_data->nat_ip, iface_vni);
+	port->iface.nat_ip = 0;
+	port->iface.nat_port_range[0] = 0;
+	port->iface.nat_port_range[1] = 0;
```

Comment:
Why is this part of the port interface now? Is that because without NAT you cannot support NAT64? (This is just so I can understand what happened, this is not a review comment)

Reply:
So an interface actually can do both NAT (ipv4) and NAT64 (for ipv6 but the NAT address is till ipv4). 
- NAT (ipv4) looks up in the hash table whether its own ipv4 source address supposed to be NATed. This approach stays the same and untouched.
- NAT64 actually decides to do NAT64 only if the target ipv6 address of the packet in a special range NAT64 range AND packet's originating interface has a NAT ip set. (port->iface.nat_ip)

This is to keep the GRPC interface backward compatible and make the NAT64 work side by side to NAT (ipv4). This is the first "backward compatible" version. We can iterate (if needed) and enhance the GRPC interface in the future to support a separate NAT IP for ipv4 and a separate NAT64 ipv4 address for ipv6. 

So `port->iface.nat_ip` is substituting the hash table lookup in SNAT table for NAT64 to answer your question in short.